versions/3.1.1.md

```diff
@@ -314,12 +314,24 @@ This is the root object of the [OpenAPI document](#openapi-description).
+| <a name="oas-security"></a>security                    |  [[Security Requirement Object](#security-requirement-object)]  | A declaration of which security mechanisms can be used across the API. See [The `security` Field](#the-security-field).                                                                                                                                                                                                                                                                                                                                                                                                                                |
 | <a name="oas-tags"></a>tags                            |                  [[Tag Object](#tag-object)]                   | A list of tags used by the document with additional metadata. The order of the tags can be used to reflect on their order by the parsing tools. Not all tags that are used by the [Operation Object](#operation-object) must be declared. The tags that are not declared MAY be organized randomly or based on the tools' logic. Each tag name in the list MUST be unique.                                                                                                                                                                               |
 | <a name="oas-external-docs"></a>externalDocs            | [External Documentation Object](#external-documentation-object) | Additional external documentation.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
 
 This object MAY be extended with [Specification Extensions](#specification-extensions).
 
+###### The `security` Field
+
+The `security` field describes how requests are authorized:
+
+- If omitted, then nothing can be inferred about the authorization requirements; the behaviour is implementation-defined.
+- If present but empty (`security: []`), then the behaviour is undefined.
```

Comment:
I think this is implementation-defined rather than undefined. There are only two viable options (which should be stated explicitly): allow-all or deny-all. It is implementation-defined as to which one of those two will happen. Undefined means that arbitrary (and arbitrarily wrong) behavior can happen, and I don't think that's a concern here. (Implementation-defined and undefined are terms that are defined within the OAS for 3.0.4 and 3.1.1, for anyone reading who was not aware of that)

Reply:
So would we change this to (paraphrasing) "it's implementation-defined, but you probably want it to be either allow all or deny all"; OR, "the behaviour MUST be either allow-all or deny-all, but which of the two, is implementation-defined" ?

Reply:
security[] == "allow all"     (+1)
I am not sure what would be the effect of making / interpreting security[] == "deny all".  Probably break some tools.

Again, I am leaning towards @handrews 's comment.  However (at least in my case) it is going to take a while to get used to security[] being in some cases deny-all if we were to go this route.

Reply:
https://github.com/OAI/OpenAPI-Specification/pull/4007#discussion_r1711137553

Sounds like there are still some pretty directly opposed viewpoints around.

Reply:
@rvedotrc 

> the behaviour MUST be either allow-all or deny-all, but which of the two, is implementation-defined

I think this would be preferable.  Pretty much everyone thinks it should be one or the other- no other options make sense.  So let's say that and leave it up to implementations which one to pick.  If we want, we can make one RECOMMENDED, but we should only do that if we are certain we will emphasize that direction in future releases.  So I'd probably avoid that for now.

Reply:
Worksforme.

Because it's not yet clear to me that there's a sufficient consensus behind any specific path forwards, before I update the PR again, let's try this:

Please +1 this comment here, if you think you would approve the PR if it said this:

 * `security` not specified == allow all
 * `security: []` MUST be either allow-all or deny-all, implementation-defined

(or -1 this comment if you think you'd request changes).

Reply:
Can't agree to the first bullet point:
* The spec currently does not define a default for this field. Adding a default is a breaking change and needs to be deferred to v4.0

Reply:
> Adding a default is a breaking change

Is it? How can you tell, given that the spec did not say what the previous behaviour was?

Reply:
> How can you tell, given that the spec did not say what the previous behaviour was?

I can tell because 
* I am omitting `security` to express "I can't tell you which authentication mechanisms will work" and 
* @miqui omits it to express "No authentication required".

Adding a default will break at least one of these interpretations.